lib/features/popup-menu/ColorMenuProvider.js

```diff
@@ -0,0 +1,63 @@
+var COLORS = [{
+  title: 'Default',
+  fill: undefined,
+  stroke: undefined
+}, {
+  title: 'Blue',
+  fill: 'rgb(187, 222, 251)',
+  stroke: 'rgb(30, 136, 229)'
+}, {
+  title: 'Orange',
+  fill: 'rgb(255, 224, 178)',
+  stroke: 'rgb(251, 140, 0)'
+}, {
+  title: 'Green',
+  fill: 'rgb(200, 230, 201)',
+  stroke: 'rgb(67, 160, 71)'
+}, {
+  title: 'Red',
+  fill: 'rgb(255, 205, 210)',
+  stroke: 'rgb(229, 57, 53)'
+}, {
+  title: 'Purple',
+  fill: 'rgb(225, 190, 231)',
+  stroke: 'rgb(142, 36, 170)'
+}];
+
+
+export default function ColorMenuProvider(popupMenu, modeling, translate) {
+  this._popupMenu = popupMenu;
+  this._modeling = modeling;
+  this._translate = translate;
+
+  this.register();
+}
+
+ColorMenuProvider.prototype.register = function() {
+  this._popupMenu.registerProvider('bpmn-color-picker', this);
+};
+
+ColorMenuProvider.prototype.getEntries = function() {
+  return [];
+};
+
+ColorMenuProvider.prototype.getHeaderEntries = function(elements) {
+  var modeling = this._modeling;
+  var translate = this._translate;
+
+  return COLORS.map(function(color) {
```

Comment:
Do we consider making the colors configurable via `config.colors` or something?

Reply:
I hadn't considered. We would need to make sure the color menu UI would support this. The menu colors are set in the CSS file at the moment, which wouldn't be possible anymore if the colors aren't fixed. And we would need to handle a variable amount of colours.

Reply:
I think it is fine to not make these variable at the moment, but also not 100% block the door regarding this.

What would be most important for me is that we have a story around "how customization is done". A number of options are on the table:

* We add the color picker per default, but make it extensible (based on what exactly?)
* We improve the existing bpmn-js-color-picker (with multi-element support); establish that as our tool for the job (and make that extensible)

Either was the result is the same: Our users can resort to a way to change colors on the diagram, both supporting single element coloring, but also grouping (new with multi-element context pad!).

I'd be absolutely fine with simply improving the existing color picker for the moment.

Reply:
I think it's OK to have it as is. Configurability can be a separate feature -> implemented via another issue.

Reply:
Generally speaking we shall not support arbitary colors; this is how our tooling works; this is how we ensure users use colors to transport semantics (their choice) but not shoot them into the foot.

Reply:
> I think it's OK to have it as is. Configurability can be a separate feature -> implemented via another issue.

But if we can go for a simple, no CSS solution to do the job, that would be great, too.

Maybe we can generate these icons as an SVG on the fly? This way we'd be able to support arbitary colors (some day maybe) but also not require CSS.

Reply:
True. I haven't noticed that the color boxes are partially defined in the stylesheet. Ideally, I'd expect the fill and stroke values to be defined in hash style due to how it is defined by the bpmn-in-color extension (`#xxxxxx`), and these values should be used to draw the icon.

Reply:
Without changes in diagram-js, we could use the `imageUrl` property on the popup entry to pass a data URL containing the colored icon: https://github.com/bpmn-io/diagram-js/blob/20ab8b869bea9dc8f7b6e9a1f98daa0eb1fe358c/lib/features/popup-menu/PopupMenu.js#L540

Reply:
Exactly exactly. :slightly_smiling_face:

Reply:
Maybe we should a proper discussion on this. Otherwise this discussion could get very long here

Reply:
Let me summarize the feedback you got:

* We should not rely on CSS for "coloring of entries", but generate images (via data-URL) and a (configurable) set of colors.
* This solves two issues: Colors should be defined in a single place; colors should be easily configurable (some day maybe).

----

Is there anything else we want to discuss? If so, let us do it.

Reply:
Proposal for configuration:

```
// disable color picker
new BpmnJS({
  colorPicker: false
});

// customize colors
new BpmnJS({
  colorPicker: {
    colors: [ { label, stroke, fill }* ]
});
```